### ISSUE TITLE
Bug: not able to serve built site in macos

### ISSUE THREAD

**Issue by unmem** [gatsbyjs/gatsby#13910]
## Description

Error when running the command `gatsby serve`.

<img width="2506" alt="Screen Shot 2019-05-08 at 12 10 47 AM" src="https://user-images.githubusercontent.com/16096559/57315297-dbfc8980-7125-11e9-91bf-009bedf79a40.png">

### Steps to reproduce
```bash
  cd /tmp
  gatsby new global-styles https://github.com/gatsbyjs/gatsby-starter-hello-world
  cd global-styles
  gatsby build
  gatsby serve
```

### Environment
    System:
      OS: macOS 10.14.4
      CPU: (6) x64 Intel(R) Core(TM) i5-8500 CPU @ 3.00GHz
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 11.13.0 - ~/.nvm/versions/node/v11.13.0/bin/node
      Yarn: 1.15.2 - /usr/local/bin/yarn
      npm: 6.7.0 - ~/.nvm/versions/node/v11.13.0/bin/npm
    Languages:
      Python: 3.7.3 - /Users/peng/.virtualenvs/python-dev/bin/python
    Browsers:
      Chrome: 74.0.3729.131
      Safari: 12.1
    npmPackages:
      gatsby: ^2.4.2 => 2.4.2

**Comment by cluzier** [comment]
You're sure you're in the correct directory? I know in the reproduction steps you do `cd tmp` but is `tmp` really the right place?

**Comment by unmem** [comment]
sorry about missing one step.. you can see the current directory in the screenshot is `global-styles`.

**Comment by polishedwp** [comment]
I can confirm on my MacOS with the Hello-world start, the same result as you have described. However, the Default starter works fine.

**Comment by thomascorthouts** [comment]
I was able to reproduce as well, a quick check learned me that adding the default `gatsby-*.js` files made this one "serve-able" from localhost. 

It seems like an empty `gatsby-config.js` is sufficient to make it work. No immediate idea on the origin of the issue, but just putting it here.

**Comment by polishedwp** [comment]
@thomascorthouts good point. the `hello-world` is a fairly blank starter and it makes since it would not `serve` in is fresh install state. 

I wonder if it would make since to include a `gatsby-config.js` file with the basic `sitemetadata` and no plugins for this starter?

**Comment by thomascorthouts** [comment]
@polishedwp indeed, it makes sense. The place it's used in the tutorial on the [website](https://www.gatsbyjs.org/tutorial/part-one/), it's only ever been run in `develop`. 

It might be a good addition to add a note that the `serve` part doesn't work in [the starter docs](https://www.gatsbyjs.org/starters/gatsbyjs/gatsby-starter-hello-world/) or as you proposed to have a   `gatsby-config.js` with basic data.

**Comment by thomascorthouts** [comment]
In version `2.3.36` of Gatsby, it still worked without a `gatsby-config.js` so it seems like a breaking change that could/should at least be resolved by adding a note about it somewhere. 
I think it might be related to the #12128 pathPrefix PR somehow, in that it expects to at least have a config file.

**Comment by thomascorthouts** [comment]
In [line 72](https://github.com/DSchau/gatsby/blob/2a761db8015f05e28fe5e212646f0287d4442c04/packages/gatsby/src/commands/serve.js#L72) of `serve.js` it is introduced to `const { pathPrefix: configPathPrefix } = config` where if config would be `null` or `undefined` it would result in a `TypeError: Cannot destructure property 'pathPrefix' of 'undefined' or 'null'.` error.

**Comment by thomascorthouts** [comment]
This issue should have been fixed with `gatsby: 2.4.3` and in the newest versions of the hello-world starter, there will also be an empty `gatsby-config.js` included. @unmem if you would run into this issue again, please feel free to open a new issue.
